src/provider/mod.rs

```diff
@@ -31,6 +34,268 @@ pub struct Eip712Fee {
     pub max_priority_fee_per_gas: u128,
 }
 
+/// Response type for `zks_getBridgeContracts`.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BridgeAddresses {
+    pub l1_shared_default_bridge: Option<Address>,
+    pub l2_shared_default_bridge: Option<Address>,
+    pub l1_erc20_default_bridge: Option<Address>,
+    pub l2_erc20_default_bridge: Option<Address>,
+    pub l1_weth_bridge: Option<Address>,
+    pub l2_weth_bridge: Option<Address>,
+    pub l2_legacy_shared_bridge: Option<Address>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BaseSystemContractsHashes {
+    pub bootloader: B256,
+    pub default_aa: B256,
+    pub evm_emulator: Option<B256>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum BlockStatus {
+    Sealed,
+    Verified,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockDetailsBase {
```

Comment:
Given that it's used in a single place via `serde(flatten)`, maybe it makes sense to move these fields to `BlockDetails` directly, or do you anticipate other use cases for this structure?

Reply:
`BlockDetails` was used in two places, but as we discussed such approach hurts the usage experience, so I removed `BlockDetails` and added corresponding fields to the two structs that were using it.